habitat_baselines/rl/ppo/policy.py

```diff
@@ -77,6 +77,10 @@ def evaluate_actions(
 
         return value, action_log_probs, distribution_entropy, rnn_hidden_states
 
+    @classmethod
+    def from_config(cls, config, envs):
```

Comment:
Actually upon review, shouldn't Policy be an abstract class like Net is in the same file? We should probably be consistent in that regard.

Reply:
Thanks for your comment!
In my humble opinion, it is considered abstract "implicitly".
I can add additional annotations as in the `Net` class if needed.
I'd be grateful to hear what other core contributors think as well.

Reply:
@erikwijmans @mathfac ^

Reply:
Updated code!

Should the `Policy`'s `forward` method be left as it is?
Neither `PointNavBaselinePolicy` nor `PointNavResNetPolicy` that inherit from `Policy` implement `forward` method.

Reply:
Yeah, let's leave forward as it is.  That raise is there because there isn't a forward method and you aren't expected to implement one (what forward should do is kinda undefined for RL so RL policies don't typically have one)

Reply:
Why do we choose `from_config` method instead of a constructor that takes `config, envs`?

Reply:
In my humble opinion, this option is more flexible. The constructor explicitly says what fields are needed to initialise the object and _from_config_ knows how to "extract" these fields from config end env and passes them to constructor.

There may be cases (hypothetically🙂) when we need to create the object (for example in tests or demo examples) "explicitly" without creating config or env.

Reply:
Explicitly makes sense, but when we pass the config to constructor of particular class that is expected to be node related to particular class and should not include unnecessary parameters. Unfortunately, that not the case here and we pass config node of higher order. That should be good to go.